component/src/main/java/org/wso2/extension/siddhi/io/ethereum/util/EthereumUtil.java

```diff
@@ -0,0 +1,193 @@
+        transactionMap.put(EthereumConstants.TRANSACTION_GAS, transaction.getGas().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_GAS_PRICE, transaction.getGasPrice().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_INDEX, transaction.getTransactionIndex().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_INPUT_STRING, transaction.getInput());
+        return transactionMap;
+    }
+
+    //map for pending transaction neglecting the null elements of a transaction.
+    private static Map<String, Object> createMapForPendingTransaction(Transaction transaction) {
+
+        Map<String, Object> transactionMap = new HashMap<>();
+        transactionMap.put(EthereumConstants.TRANSACTION_HASH, transaction.getHash());
+        transactionMap.put(EthereumConstants.TRANSACTION_SENDER, transaction.getFrom());
+        transactionMap.put(EthereumConstants.TRANSACTION_RECIPIENT, transaction.getTo());
+        transactionMap.put(EthereumConstants.TRANSACTION_AMOUNT, transaction.getValue().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_GAS_PRICE, transaction.getGasPrice().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_GAS, transaction.getGas().intValue());
+        transactionMap.put(EthereumConstants.TRANSACTION_INPUT_STRING, transaction.getInput());
+        return transactionMap;
+    }
+
+    //map for new block
+    private static Map<String, Object> createMapForNewBlock(EthBlock.Block block) {
+
+        Map<String, Object> blockMap = new HashMap<>();
+        blockMap.put(EthereumConstants.BLOCK_HASH, block.getHash());
+        blockMap.put(EthereumConstants.BLOCK_NUMBER, block.getNumber().intValue());
+        blockMap.put(EthereumConstants.BLOCK_DIFFICULTY, block.getDifficulty().intValue());
+        blockMap.put(EthereumConstants.BLOCK_TOTAL_DIFFICULTY, block.getTotalDifficulty().intValue());
+        blockMap.put(EthereumConstants.BLOCK_GAS_LIMIT, block.getGasLimit().intValue());
+        blockMap.put(EthereumConstants.BLOCK_GAS_USED, block.getGasUsed().intValue());
+        blockMap.put(EthereumConstants.BLOCK_PARENT_HASH, block.getParentHash());
+        blockMap.put(EthereumConstants.BLOCK_MIX_HASH, block.getMixHash());
+        blockMap.put(EthereumConstants.BLOCK_NONCE, block.getNonce().intValue());
+        blockMap.put(EthereumConstants.BLOCK_SIZE, block.getSize().intValue());
+        blockMap.put(EthereumConstants.BLOCK_MINER, block.getMiner());
+        blockMap.put(EthereumConstants.BLOCK_RECEIPT_ROOT, block.getReceiptsRoot());
+        blockMap.put(EthereumConstants.BLOCK_STATE_ROOT, block.getStateRoot());
+        blockMap.put(EthereumConstants.BLOCK_TRANSACTIONS_ROOT, block.getTransactionsRoot());
+        blockMap.put(EthereumConstants.BLOCK_LOGS_BLOOM, block.getLogsBloom());
+        blockMap.put(EthereumConstants.BLOCK_SHA3_UNCLES, block.getSha3Uncles());
+        return blockMap;
+    }
+
+    // to create filtering
```

Comment:
Better to add comments like [this](https://github.com/siddhi-io/siddhi-io-http/blob/master/component/src/main/java/io/siddhi/extension/io/http/source/HttpRequestSource.java#L548)